### ISSUE TITLE
Query fail with error `failed: at index 1`

### ISSUE THREAD

**Issue by malwaregarry** [prestodb/presto#23613]
## Your Environment

* Presto version used: 0.288 docker image
* Storage (HDFS/S3/GCS..):
* Data source and connector used: Memory connector
* Deployment (Cloud or On-prem): On-prem
* [Pastebin](https://pastebin.com/) link to the complete debug logs: https://pastebin.com/Duaea2PE

## Expected Behavior

Should  return true or provide a more meaningful error message

## Current Behavior

returns error `Query 20240910_100645_00342_v3b4v failed: at index 1`

## Steps to Reproduce
The select query returns an error:
```
select (CASE (CASE true WHEN (true) THEN (true) ELSE coalesce((false = ANY (VALUES true)), (true)) END ) WHEN false THEN true END );

```

The subquery  returns true:
```
select CASE true WHEN (true) THEN (true) ELSE coalesce((false = ANY (VALUES true)), (true)) END ;

```

Changing `false = ANY (VALUES true)), false` also results in no errors:
```
select (CASE (CASE true WHEN (true) THEN (true) ELSE coalesce((false), (true)) END ) WHEN false THEN true END );

```

## Context
Test case generated from SQLancer

**Comment by tdcmeehan** [comment]
@malwaregarry Did you post the right failure query?  I am unable to reproduce, and I think all 3 queries you posted are identical.

**Comment by malwaregarry** [comment]
@tdcmeehan sorry the 2nd and 3rd queries were incorrect, but the first one should cause an error.

This is the command I used to start presto
`docker run -p 8080:8080 -d -v ./memory.properties:/opt/presto-server/etc/catalog/memory.properties --name presto prestodb/presto:0.288`

with the memory.property file having:
```
connector.name=memory
memory.max-data-per-node=128MB
```

**Comment by tdcmeehan** [comment]
#23629 fixes this.  Thanks for the report @malwaregarry!

**Comment by tdcmeehan** [comment]
Fixed in #23629
